Dungeoneer/src/com/interrupt/dungeoneer/gfx/GlRenderer.java

```diff
@@ -778,6 +780,117 @@ public void updateShaderAttributes() {
 		}
 	}
 
+	TextureRegion tempGlyphTextureRegion = new TextureRegion();
+	public void renderTextBatches() {
+		DecalBatch batch = getDecalBatch("sprite", Entity.BlendMode.OPAQUE);
+		if (renderingForPicking)
+			batch = getDecalBatch("picking", Entity.BlendMode.OPAQUE);
+
+		// Global text scale modifier. Make a constant?
+		float baseTextScale = 0.025f;
+
+		Color tempColor = new Color();
+
+		for (int i = 0; i < textToRender.size; i++) {
+			DrawableText dT = textToRender.get(i);
+
+			float curXPos = 0f, curZPos = 0f;
+
+			// TODO: Handle newline characters?
+
+			GlyphLayout bounds = FontBounds.GetBounds(font, dT.text);
+			float textWidth = bounds.width * dT.scale * baseTextScale;
+			float textHeight = bounds.height * dT.scale * baseTextScale;
+
+			BitmapFont.Glyph glyph = font.getData().getGlyph('X');
+
+			float glyphWidth, glyphHeight = glyph.height * dT.scale * baseTextScale;
+
+			// Draw a decal per-glyph for this text
+			for(int ii = 0; ii < dT.text.length(); ii++) {
+				char character =  dT.text.charAt(ii);
+
+				glyph = font.getData().getGlyph(character);
+
+				if (glyph == null && character == '\n') { // Newline support is in DrawableText, replaces "\\n" with "\n" there to avoid issues with font boundary calculations.
+                    curXPos = 0;
+                    curZPos += glyphHeight;
+                    continue;
+				}
+
+				glyphWidth = glyph.width * dT.scale * baseTextScale;
+
+				float tx = dT.parentPosition.x + curXPos;
+				float ty = dT.parentPosition.y + 0.001f; // Pull out a bit, to place directly on walls
+				float tz = dT.parentPosition.z - curZPos + (glyphHeight * 0.5f); // Place font baseline directly on entity origin
+
+				// Center text on origin
```

Comment:
Currently the `Text` entity origin is aligned with the first character. This may introduce off pixel alignment when snapping?

Reply:
`NeoText` aligns on the entity origin, which I think makes for more sensible rotations and world 'presence.' Please review the snapping features for behaviour.